includes/notes/class-wc-payments-notes-instant-deposits-eligible.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Notify merchant that they are eligible for Instant Deposits.
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Payments_Notes_Set_Https_For_Checkout
+ */
+class WC_Payments_Notes_Instant_Deposits_Eligible {
+	use NoteTraits {
+		can_be_added as protected trait_can_be_added;
+	}
+
+	/**
+	 * The transient that the account data is stored in.
+	 */
+	const ACCOUNT_TRANSIENT = 'wcpay_account_data';
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-payments-notes-instant-deposits-eligible';
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	// TODO: Get the proper doc url.
+	const NOTE_DOCUMENTATION_URL = '';
+
+	/**
+	 * Checks if a note can and should be added.
+	 *
+	 * @return bool
+	 */
+	public static function can_be_added() {
+		$account = get_transient( self::ACCOUNT_TRANSIENT );
```

Comment:
At this point, there may be nothing in the transient. It's better to use `WC_Payments_Account->get_cached_account_data()`, which will take care of error handling and re-fetching the data from the server if the transient has expired.

Reply:
I reworked this so that the test is no longer in the note itself, it lives in the code that adds the note.